src/Workspaces/Core/Portable/Workspace/ProjectSystem/IHostDiagnosticAnalyzerProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Workspaces.ProjectSystem
+{
+    /// <summary>
+    /// An interface implemented by hosts to provide the host-level analyzers; for example in Visual Studio for Windows this
+    /// is where we'll fetch VSIX-defined analyzers.
+    /// </summary>
+    internal interface IHostDiagnosticAnalyzerProvider
```

Comment:
do we need the factory as an interface?  
https://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Diagnostics/VisualStudioDiagnosticAnalyzerProvider.Factory.cs,19

Or what is the way we provide the implementation?

Reply:
It might be a MEF part, but just grabbed and passed in directly to the project type.